src/cross_section.F90

```diff
@@ -153,87 +167,119 @@ subroutine calculate_nuclide_xs(i_nuclide, i_sab, E, i_mat, i_nuc_mat, i_log_uni
     nuc => nuclides(i_nuclide)
     mat => materials(i_mat)
 
-    ! Determine index on nuclide energy grid
-    select case (grid_method)
-    case (GRID_MAT_UNION)
+    ! If MP, don't interpolate, it's all already baked in.
+    if( nuc % mp_present .AND. &
```

Comment:
For spacing, it should look like `if (nuc % mp_present .and. &`. See the [relevant section](http://openmc.readthedocs.org/en/latest/devguide/styleguide.html#whitespace-in-expressions) in the style guide.